### ISSUE TITLE
Missing optimization when fusion definition is using div(a,b) instead of mul(a,reciprocal(b))

### ISSUE THREAD

**Issue by liqiangxl** [NVIDIA/Fuser#642]
When benchmarking `softmax` noticed about 10% performance drop if fusion definition is using div(a,b) instead of mul(a,reciprocal(b)) . The `mul` version avoided multiple `div` ops in a for-loop.
```
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[-1, -1], contiguity=[True, True], dtype=DataType.Half, is_cpu=False)
    T1 = fd.ops.cast(T0, dtype=DataType.Float)
    T2 = fd.ops.max(T1, axes=[1], keepdim=True, dtype=DataType.Null)
    T5 = fd.ops.sub(T1, T2)
    T6 = fd.ops.exp(T5)
    T7 = fd.ops.sum(T6, axes=[1], keepdim=True, dtype=DataType.Null)
    T10 = fd.ops.div(T6, T7)
    T11 = fd.ops.cast(T10, dtype=DataType.Half)
    fd.add_output(T11)

with FusionDefinition() as fd:
    nvfuser_fusion_id0(fd)

inputs = [
    torch.randn((8192, 10240), dtype=torch.float16, device='cuda:0').as_strided((8192, 10240), (10240, 1)),
]
fd.execute(inputs)
```

Optimized should be:
```
  T8 = fd.ops.reciprocal(T7)  
  T10 = fd.ops.mul(T6, T8)
```
